src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -20,6 +20,50 @@
 namespace Microsoft.PowerShell.Commands
 {
     /// <summary>
+    /// Exception class for webcmdlets to enable returning HTTP error response
+    /// </summary>    
+    public class HttpResponseException : Exception
+    {
+        private HttpResponseMessage _response;
+        private HttpStatusCode _status;
+        private HttpResponseHeaders _headers;
+
+        /// <summary>
+        /// Constructor for HttpResponseException
+        /// </summary>
+        /// <param name="message">Message for the exception</param>
+        public HttpResponseException (string message) : base(message)
+        {}
+
+        /// <summary>
+        /// HTTP error response
+        /// </summary>
+        public HttpResponseMessage Response
+        {
+            get { return _response; }
+            set { _response = value; }
+        }
+
+        /// <summary>
+        /// HTTP error status code
+        /// </summary>
+        public HttpStatusCode Status
```

Comment:
Was originally adding the individual properties before I saw the example of Windows PowerShell just having a Response property.  Didn't remove them.  Will remove.